src/app/file-manager/services/register-download.service.ts

```diff
@@ -1,16 +1,33 @@
+
 import { Injectable } from '@angular/core';
 import { ElectronService } from 'ngx-electron';
 import { SecurityService } from './security.service';
 import { Item } from '../models/item';
+import { Response } from '@angular/http/src/static_response';
+import { DiskStatus } from '../models/diskStatus';
 
 @Injectable()
 export class RegisterDownloadService {
-
   constructor(private electronService: ElectronService) { }
+  public diskStatus: DiskStatus;
 
   public startDownload(items: Item[]) {
     if (this.electronService.isElectronApp) {
       this.electronService.ipcRenderer.send('start-download', items, SecurityService.getAccessToken());
     }
   }
+
+  public verifyDisk(destination: String, totalfilesSize: Number): Promise<DiskStatus> {
+    return new Promise(result => {
+
+      this.electronService.ipcRenderer.removeAllListeners('return-disk-response');
```

Comment:
I don't think this is the correct way to have an IPC that sends a response back; you shouldn't need another IPC message for a response. It looks like using a Promise would work here: https://www.npmjs.com/package/electron-ipc

Reply:
This library hasn't been updated for more than a year, and we cannot access its code from github, so we think it isn't convenient to implement this.

Reply:
Oh, I thought this was part of electron, not an add-on. Can you use `ipcRender.once()` to avoid having to remove the listener?

Reply:
Oh yeah! `ipcRender.once()` works smoothly. Nice! I'll update this change.